### PR TITLE
remove obsolete references to the product type

### PR DESCRIPTION
**Pull request by nickpell**
This should always be CommCare going forward, for all UIs and new model instances.

🐡 or 🏢 

@proteusvacuum / @benrudolph cc: @snopoke

### CODE REVIEW

File: corehq/apps/domain/views.py [hunk 1]
```diff
@@ -632,7 +632,7 @@ def dispatch(self, request, *args, **kwargs):
     @property
     @memoized
```
Comment (snopoke) [inline corehq/apps/domain/views.py]: don't need to memoize this anymore

**Comment by proteusvacuum** [comment]
:+1:

File: corehq/apps/accounting/models.py [hunk 2]
```diff
@@ -1455,24 +1441,21 @@ def send_ending_reminder_email(self):
         user_desc = self.plan_version.user_facing_description
         plan_name = user_desc['name']
         domain_name = self.subscriber.domain
-        product = self.plan_version.core_product
         emails = {a.username for a in WebUser.get_admins_by_domain(domain_name)}
         emails |= {e for e in WebUser.get_dimagi_emails_by_domain(domain_name)}
         if self.is_trial:
-            subject = _("%(product)s Alert: 30 day trial for '%(domain)s' "
+            subject = _("CommCare Alert: 30 day trial for '%(domain)s' "
```
Comment (benrudolph) [inline corehq/apps/accounting/models.py]: not your code, but this'll crash on unicode domains right?

**Comment by benrudolph** [comment]
much simpler 💯

**Pull request merged**